Examples/Algorithms/TruthTracking/ActsExamples/TruthTracking/ParticleSmearing.cpp

```diff
@@ -106,6 +106,9 @@ ActsExamples::ProcessCode ActsExamples::ParticleSmearing::execute(
           phi + sigmaPhi * stdNormal(rng), theta + sigmaTheta * stdNormal(rng));
       params[Acts::eBoundPhi] = newPhi;
       params[Acts::eBoundTheta] = newTheta;
+      if (std::abs(newTheta) < 1e-6) {
+        throw std::runtime_error("Theta is too close to zero.");
+      }
```

Comment:
I don't think this is a good solution because ultimately the cutoff should depend on the numerical precision so if we want to put in a value we have to experiment with the cutoff and should declare it somewhere central. I think there is actually a similar value for the `z` component.

Apart from that `theta` is inside `[0, pi]` so this expression would only check the positive z axis right now.